timestamp.py

```diff
@@ -87,9 +89,32 @@ def file():
 def downloadCert():
     text = request.form['data']
     plaintext_bytes = text.encode('utf-8')
-    with open(public, "r") as myfile:
-        pub_key = RSA.importKey(myfile.read())
-    encrypted = encrypt(plaintext_bytes,  pub_key )
+    pub_key = False
+    try:
+        with open(public, "rb") as myfile:
+            private_key = RSA.importKey(myfile.read())
+    except:
+        #Signature, change public and private keys so that
+        #we can encrypt with private and decrypt with public...
+        #Due to implementation reasons. In general, modul does not
+        #agree on encrypting with (real RSA) private key.
+
+        #We just publish RSA private key insted of public 
+        # #-- not so goot, but works for now.
+        #publish one for verification
+        key = RSA.generate(2048)
+        f = open(public,'wb')
+        f.write(key.exportKey('PEM'))
+        f.close()
+        #store one for signing
+        f = open(private,'wb')
+        f.write(key.publickey().exportKey('PEM'))
```

Comment:
The web app should not write anything on the disk - in fact, the corresponding files are marked as read-only on the server. This code can go to a separate set-up helper script, while the main application should just assume that everything has been set up properly.